examples/src/main/java/com/hedera/hashgraph/sdk/examples/MirrorNodeContractQueriesExample.java

```diff
@@ -0,0 +1,193 @@
+            .setMaxTransactionFee(Hbar.from(2))
+            .execute(client);
+
+        TransactionReceipt fileCreateTxReceipt = fileCreateTxResponse.getReceipt(client);
+        FileId newFileId = fileCreateTxReceipt.fileId;
+
+        Objects.requireNonNull(newFileId);
+        System.out.println("Created new file with ID: " + newFileId);
+
+        var response = new ContractCreateTransaction()
+            .setAdminKey(client.getOperatorPublicKey())
+            .setGas(200000)
+            .setConstructorParameters(new ContractFunctionParameters().addString("Hello from Hedera."))
+            .setBytecodeFileId(newFileId)
+            .setContractMemo("Simple contract with string field")
+            .execute(client);
+
+        var contractId = Objects.requireNonNull(response.getReceipt(client).contractId);
+
+        /*
+         * Step 3:
+         * Wait for mirror node to import data
+         */
+        Thread.sleep(4000);
+
+        /*
+         * Step 4:
+         * Estimate the gas needed
+         */
+        var gas = new MirrorNodeContractEstimateGasQuery()
+            .setContractId(contractId)
+            .setSender(client.getOperatorAccountId())
+            .setGasLimit(30_000)
+            .setGasPrice(1234)
+            .setFunction("getMessage")
+            .execute(client);
+
+        System.out.println("Gas needed for this query: " + gas);
+
+        /*
+         * Step 5:
+         * Do the query against the consensus node using the estimated gas
+         */
+        var callQuery = new ContractCallQuery()
+            .setContractId(contractId)
+            .setGas(gas)
+            .setFunction("getMessage")
+            .setQueryPayment(new Hbar(1));
+
+        var result = callQuery
+            .execute(client);
+
+        /*
+         * Step 6:
+         * Simulate the transaction for free, using the mirror node
+         */
+        var simulationResult = new MirrorNodeContractCallQuery()
+            .setContractId(contractId)
+            .setSender(client.getOperatorAccountId())
+            .setGasLimit(30_000)
+            .setBlockNumber(10000)
+            .setGasPrice(1234)
+            .setFunction("getMessage")
+            .execute(client);
+
+        System.out.println("Simulation result: " + decodeABIHexString(simulationResult));
+        System.out.println("Contract call result: " + result.getString(0));
+    }
+    private static String decodeABIHexString(String hex) {
+        // Trim 0x at the beginning
+        if (hex.startsWith("0x")) {
+            hex = hex.substring(2);
+        }
+
+        // Extract the length of the data by parsing the substring from position 64 to 128 as a hexadecimal integer
+        // This section represents the length of the dynamic data, specifically the number of bytes in the string or array
+        int length = Integer.parseInt(hex.substring(64, 128), 16);
```

Comment:
Good logic but it may not be clear to all devs why the indexes at 64, 128 are important.
You should add a function comment that basically gives users an overview of your algorithm so they can understand why your example takes this approach

Reply:
Added javadoc